### DISCUSSION TITLE
Newbie questions regarding updating the offline and online store

### DISCUSSION THREAD

**Discussion by RoyRubinTLVMC** [feast-dev/feast#4313] [Q&A] [answered]
Hi, after reading the docs, and the quick start guide
I did not quite understand how the offline store is updated, and how it affects the online store.

Here is my simple testing use case:

I have data stored in several databases.
Every few hours I generate features from these databases.
the features will be stored in a parquet file. 
this means - a new parquet file will be generated every few hours with the latest features.
after latest features are generated, I want to use them for model training (offline) or serving (online).

from the examples in the tutorial, it seemed to me that the feast "apply" command is performed only once.

1) how can my offline store be updated with the latest information ? do I have to do feast "apply" manually each time a parquet file is created ?
2) how does the online store adjust accordingly ? does it simply read the "latest" data from the offline store, and stashes it in the online database ? if the offline store receives new features - does the online store update automatically ?

The answer is probably hiding somewhere in the docs, but I couldn't find it.

Thanks in advance !

**Comment by franciscojavierarceo** [comment, marked as answer]
>1. how can my offline store be updated with the latest information ? do I have to do feast "apply" manually each time a parquet file is created ?
Include it in the path in the `SparkSource` object. Here's some [more documentation](https://docs.feast.dev/reference/data-sources/spark).

> 2. how does the online store adjust accordingly ? does it simply read the "latest" data from the offline store, and stashes it in the online database ? if the offline store receives new features - does the online store update automatically ?
You have to explicitly materialize the data to the online store. [See this documentation](https://docs.feast.dev/reference/batch-materialization/spark).
